src/Uno.UI/UI/Xaml/Controls/Frame/Frame.partial.mux.cs

```diff
@@ -48,6 +49,17 @@ protected override void OnApplyTemplate()
 
 		base.OnApplyTemplate();
 
+#if HAS_UNO
+		if (this.TemplatedRoot?.GetType().Name?.Contains("NativeFramePresenter", StringComparison.Ordinal) == true)
+		{
+			// It is not possible to use the WinUI behavior with a NativeFramePresenter.
+			if (this.Log().IsEnabled(LogLevel.Error))
+			{
+				this.Log().LogError("WinUI Frame behavior is not compatible with NativeFramePresenter. Set the Frame.Style to '{StaticResource XamlDefaultFrame}' instead.");
```

Comment:
Should we perhaps "undo" the feature flags for NatveFramePresenter if `UseWinUIBehavior` for `Frame` is set?

https://github.com/unoplatform/uno/blob/4babcb40873138e6108b387d4732506c02c26824/src/Uno.UI/FeatureConfiguration.cs#L470C23-L470C53

Reply:
The trouble here is that in case of the Toolkit `NativeFramePresenter` I don't think they are set - the style coming from toolkit is set always irrespective of the flags, unless I missed something. If that is the case, then I think this solution will apply to both cases.

Secondly, you could mix WinUI and non-WinUI behavior frames in a single apps, in which case this hint may be better.